src/Common/src/System/Net/Security/NegotiateStreamPal.Unix.cs

```diff
@@ -98,13 +98,26 @@ internal static string QueryContextAuthenticationPackage(SafeDeleteContext secur
             ref SafeGssContextHandle context,
             SafeGssCredHandle credential,
             bool isNtlm,
+            SecurityBuffer cbt,
             SafeGssNameHandle targetName,
             Interop.NetSecurityNative.GssFlags inFlags,
             byte[] buffer,
             out byte[] outputBuffer,
             out uint outFlags,
             out int isNtlmUsed)
         {
+            // If a TLS channel binding token (cbt) is available then get the pointer
+            // to the application specific data.
+            IntPtr cbtAppData = IntPtr.Zero;
+            int cbtAppDataSize = 0;
+            if (cbt != null)
+            {
+                int appDataOffset = Marshal.SizeOf<SecChannelBindings>();
+                Debug.Assert(appDataOffset < cbt.size);
+                cbtAppData = cbt.unmanagedToken.DangerousGetHandle() + appDataOffset;
```

Comment:
Something's keeping the SafeHandle alive / not disposed such that this DangerousGetHandle is actually safe?  Or we just know that by construction?

Reply:
I considered that.  It's alive by the upstack callers and it's only used for a brief period of time to calculate the hash of the NTLM token.  So, it doesn't go out of scope etc.